deployment/morph/playgrounds.nix

```diff
@@ -0,0 +1,133 @@
+{
+  mkInstance = { defaultMachine, plutus-playground, marlowe-playground, pkgs, ports, machines, ... }:
+    hostName:
+    let
+      promNodeTextfileDir = pkgs.writeTextDir "roles.prom"
+        ''
+          machine_role{role="playgrounds"} 1
+        '';
+      plutusPlayground = "${plutus-playground.server-invoker}/bin/plutus-playground";
+      marlowePlayground = "${marlowe-playground.server-invoker}/bin/marlowe-playground";
+      playgroundServiceConfig = {
+        TimeoutStartSec = "0";
+        Restart = "always";
+        DynamicUser = true;
+        StateDirectory = [ "pab" ];
+        ProtectKernelTunables = true;
+        ProtectControlGroups = true;
+        ProtectKernelModules = true;
+        PrivateDevices = true;
+        SystemCallArchitectures = "native";
+        ConfigurationDirectory = "playground";
+      };
+      killallz3 = pkgs.writeScriptBin "killallz3" ''
+        kill -9 $(ps aux | grep z3 | grep -v grep | awk '{print $2}')
+      '';
+    in
+    { config, pkgs, lib, ... }:
+    {
+      deployment.secrets = {
+        "plutus-secrets" = {
+          source = "./secrets.plutus.${machines.environment}.env";
+          destination = "/etc/playground/secrets.plutus.${machines.environment}.env";
+          action = [ "systemctl" "restart" "plutus-playground" ];
+        };
+        "marlowe-secrets" = {
+          source = "./secrets.marlowe.${machines.environment}.env";
+          destination = "/etc/playground/secrets.marlowe.${machines.environment}.env";
+          action = [ "systemctl" "restart" "marlowe-playground" ];
+        };
+      };
+      imports = [ (defaultMachine hostName pkgs) ];
+
+      networking.firewall = {
+        enable = true;
+        allowedTCPPorts = with ports; [ ssh plutus-playground-webserver marlowe-playground-webserver nodeExporter ];
+      };
+
+
+      systemd.services.plutus-playground = {
+        enable = true;
+        after = [ "network.target" ];
+        wantedBy = [ "nginx.service" ];
+        before = [ "nginx.service" ];
+        serviceConfig = playgroundServiceConfig // {
+          EnvironmentFile = "/etc/playground/secrets.plutus.${machines.environment}.env";
+        };
+        script = "${plutusPlayground} webserver -p 4000";
```

Comment:
Shouldn't the `4000` here be something from `ports.plutus-playground` ?

Reply:
its local but I can add it to a `let`